src/panels/lovelace/editor/config-elements/hui-map-card-editor.ts

```diff
@@ -83,8 +67,68 @@ export class HuiMapCardEditor extends LitElement implements LovelaceCardEditor {
 
   @state() private _configEntities?: EntityConfig[];
 
+  private _schema = memoizeOne(
+    (localize: LocalizeFunc) =>
+      [
+        { name: "title", selector: { text: {} } },
+        {
+          name: "",
+          type: "expandable",
+          iconPath: mdiPalette,
+          title: localize(`ui.panel.lovelace.editor.card.map.appearance`),
+          schema: [
+            {
+              name: "",
+              type: "grid",
+              schema: [
+                { name: "aspect_ratio", selector: { text: {} } },
+                {
+                  name: "default_zoom",
+                  default: DEFAULT_ZOOM,
+                  selector: { number: { mode: "box", min: 0 } },
+                },
+                {
+                  name: "theme_mode",
```

Comment:
Shouldn't we just leave this out of the UI and promote the use of the theme variable `map-filter`?

Reply:
If you say this option should be removed, and build it all with themes, I can understand that argument. I don't know if I fully agree with it, as I think that is pretty complicated to ask users to do, but I understand not wanting to maintain multiple solutions. In the past I have added theme vars to solve problems and I get a lot of "what's a theme? how do I add what where? help I broke my config", etc. So I think many are not too familiar with it, it's somewhat advanced. The map filter string is pretty intimidating as well. 

If we did want to replicate this functionality with themes I think we at least need one new theme var, to set the color of the map icons, and the color of the Reset Focus button. If overall map background color is independently themable then those need to be independently themable as well to not clash.

So that is one decision to make. 

However if we do decide to keep this `theme_mode` as an option, I'm not sure what the point is to build it, but then hide it by removing from the UI. Just seems kind of user-hostile for no reason.


Anyway let me know what you think so we can try to align on that and get this issue closed out. Thanks.

Reply:
I guess one of the issues I have is that once you set a `theme_mode` you can no longer theme the color of the controls, as they are hard coded then.

But I do agree that it is not easy to create a theme for things like this, but ideally a user using a community created theme doesnt have to change the config of cards to get it to match the theme.

Not sure what the right approach her would be.